src/test/java/com/puppycrawl/tools/checkstyle/checks/modifier/RedundantModifierCheckTest.java

```diff
@@ -231,6 +231,26 @@ public void testEnumMethods() throws Exception {
         verify(checkConfig, getPath("InputRedundantModifierFinalInEnumMethods.java"), expected);
     }
 
+    @Test
+    public void testEnumMethodsInFinalClass() throws Exception {
+        final DefaultConfiguration checkConfig = createModuleConfig(RedundantModifierCheck.class);
+        final String[] expected = {
+            "12:23: " + getCheckMessage(MSG_KEY, "final"),
+        };
+        verify(checkConfig,
+            getPath("InputRedundantModifierFinalInEnumMethodsInFinalClass.java"), expected);
+    }
+
+    @Test
+    public void testEnumStaticMethodsInPublicClass() throws Exception {
+        final DefaultConfiguration checkConfig = createModuleConfig(RedundantModifierCheck.class);
+        final String[] expected = {
+            "12:23: " + getCheckMessage(MSG_KEY, "final"),
+        };
+        verify(checkConfig,
+            getPath("InputRedundantModifierFinalInEnumStaticMethods.java"), expected);
```

Comment:
`InputRedundantModifierFinalInEnumMethodsInFinalClass` is a final class which lead to this issue.

`InputRedundantModifierFinalInEnumStaticMethods` is a class without `final` modifier test for [the new violation](https://djydewang.github.io/diffGroovyReport_Issue5268/elasticsearch/xref/home/bbg/project/contribution/checkstyle-tester/repositories/elasticsearch/src/main/java/org/elasticsearch/transport/Transports.java.html#L35).

Reply:
@djydewang I see, I missed this one detail.
Why does the `final` class matter when it doesn't produce a different violation? Was this required for code coverage or pitest or another CI?

Reply:
@rnveach 
> Why does the final class matter when it doesn't produce a different violation?

It does no matter now but it produce a different violation in old version.
> Was this required for code coverage or pitest or another CI?

No.

You can see the detail of adding InputRedundantModifierFinalInEnumStaticMethods at [conversation](https://github.com/checkstyle/checkstyle/pull/5301#issuecomment-347833988).

Reply:
> InputRedundantModifierFinalInEnumStaticMethods 

Yes, what but what about `InputRedundantModifierFinalInEnumMethodsInFinalClass.java`? I don't see anything in regression yet that said `final` in class with nested `enum` caused a difference.

Reply:
InputRedundantModifierFinalInEnumMethodsInFinalClass.java comes from description of this issue:)

Reply:
@djydewang Since we have `InputRedundantModifierFinalInEnumStaticMethods` which closely matches the issue and definitely matches regression, I think we can remove `InputRedundantModifierFinalInEnumMethodsInFinalClass`.

Reply:
OK~I get it.
Thanks for your advice.

Reply:
@rnveach 
Done.